ts/packages/node/src/stores/IniFile.ts

```diff
@@ -0,0 +1,55 @@
+import { promises as fs } from 'fs';
+import _ from 'lodash';
+import ini from 'ini';
+import { ConfigStore, ConfigStoreQuery, Config } from '@configu/ts';
+
+export type IniFileConfigStoreConfiguration = { path: string };
+
+export class IniFileConfigStore extends ConfigStore {
+  private readonly path: string;
+  constructor({ path }: IniFileConfigStoreConfiguration) {
+    super('ini-file');
+    this.path = path;
+  }
+
+  async read(): Promise<Config[]> {
+    const data = await fs.readFile(this.path, 'utf8');
+    const iniObject = ini.parse(data);
+
+    return Object.entries(iniObject).flatMap(([set, keyValuePairs]) =>
+      Object.entries(keyValuePairs)
+        .filter(([, value]) => typeof value === 'string' && !Array.isArray(value))
+        .map(([key, value]) => ({ set, key, value: value as string })),
+    );
+  }
+
+  async write(nextConfigs: Config[]): Promise<void> {
+    const iniObject = nextConfigs.reduce<Record<string, Record<string, string>>>((acc, config) => {
+      const set = config.set || '';
+      if (!acc[set]) {
+        acc[set] = {};
+      }
+      acc[set]![config.key] = config.value;
+      return acc;
+    }, {});
```

Comment:
This logic should take into consideration that key-value pairs that belong to the root set need to appear before any other section.

Here is a suggestion for such an implementation:
```suggestion
    const groupedConfigs = _(nextConfigs)
      .groupBy('set')
      .mapValues((setConfigs) => _.merge({}, ...setConfigs.map((config) => ({ [config.key]: config.value }))))
      .value();
    const rootConfigs = groupedConfigs[''];
    const iniObject = _.merge(rootConfigs, _.omit(groupedConfigs, ''));
```